model/publishing/delivery/tasks/DeployTestEnvironments.php

```diff
@@ -134,14 +138,39 @@ protected function compileTest(\core_kernel_classes_Resource $env, \core_kernel_
                     );
                     $this->addChildId($taskId);
                 }
-                $report = new \common_report_Report(\common_report_Report::TYPE_SUCCESS, __('Test has been compiled as %s', $delivery->getUri()));
+                $report = new common_report_Report(common_report_Report::TYPE_SUCCESS, __('Test has been compiled as %s', $delivery->getUri()));
                 $report->setData($delivery->getUri());
+
                 return $report;
             } else {
-                return new \common_report_Report(\common_report_Report::TYPE_ERROR, __('Failed to compile %s with message: %s', $test, $response->getBody()->getContents()));
+                return new common_report_Report(common_report_Report::TYPE_ERROR, __('Failed to compile %s with message: %s', $test, $response->getBody()->getContents()));
             }
-        } catch (\Exception $e) {
-            return new \common_report_Report(\common_report_Report::TYPE_ERROR, __('Failed to compile %s with message: %s', $test, $e->getMessage()));
+        } catch (Exception $e) {
+            return new common_report_Report(common_report_Report::TYPE_ERROR, __('Failed to compile %s with message: %s', $test, $e->getMessage()));
         }
     }
+
+    /**
+     * @param core_kernel_classes_Resource $delivery
+     * @return File
+     * @throws \core_kernel_persistence_Exception
+     */
+    private function getQtiTestPackageFile(core_kernel_classes_Resource $delivery): File
+    {
+        $packagePath = $delivery->getOnePropertyValue($this->getProperty(TestBackupService::PROPERTY_QTI_TEST_BACKUP_PATH));
+        $packagePath = (string)  $packagePath;
+        
+        return $this->getServiceLocator()
+            ->get(FileSystemService::SERVICE_ID)
+            ->getDirectory(TestBackupService::FILESYSTEM_ID)
+            ->getFile($packagePath);
```

Comment:
The test backup file might be missing. Maybe for fallback it could be exported on the fly?

Reply:
As we discussed at the moment there is no possibility to publish an already existing delivery, so it's won't be a breaking change. We will add error message in report log why publication to remote environment failed.